Reset state fails with time type

The following program fails to compile, apparently because the reset state t has type `time`.  If you change its type to `int`, everything works.  The error message is:

```
/Users/eal/git/lf-3pi-template/src-gen/ModalBug/ModalBug.c:85:98: error: cannot take the address of an rvalue of type 'long long'
        envs[modalbug_main].modes->state_resets[modal_state_reset_count[modalbug_main]].source = &SEC(0);
                                                                                                 ^~~~~~~
1 error generated.
gmake[2]: *** [CMakeFiles/ModalBug.dir/build.make:118: CMakeFiles/ModalBug.dir/ModalBug.c.o] Error 1
```

Note the erroneous `&SEC(0)`.

```
target C
reactor C {
  input trigger:bool
  reset state t:time = 0s
  reaction(trigger) {=
    lf_print("t = %d", self->t);
    self->t = lf_time_logica();
  =}
}
main reactor {
  timer t(0, 1s)
  initial mode A {
    c = new C()
    reaction(t) -> reset(B), c.trigger {=
      lf_print("In A");
      lf_set(c.trigger, true);
      lf_set_mode(B);
    =}
  }
  mode B {
    reaction(t) -> reset(A) {=
      lf_print("In B");
      lf_set_mode(A);
    =}
  }
}
```

--- a/core/src/main/java/org/lflang/generator/c/CStateGenerator.java
+++ b/core/src/main/java/org/lflang/generator/c/CStateGenerator.java
@@ -88,8 +88,7 @@ public class CStateGenerator {
             + "]";
     var type = types.getTargetType(instance.tpr.resolveType(ASTUtils.getInferredType(stateVar)));
 
-    if (ASTUtils.isOfTimeType(stateVar)
-        || ASTUtils.isParameterized(stateVar) && !stateVar.getInit().getExprs().isEmpty()) {
+    if (ASTUtils.isParameterized(stateVar) && !stateVar.getInit().getExprs().isEmpty()) {
       return CModesGenerator.generateStateResetStructure(
           instance, modeRef, selfRef, stateVar.getName(), initExpr, type);
     } else {

--- a/test/C/src/modal_models/ResetStateVariableOfTypeTime.lf
+++ b/test/C/src/modal_models/ResetStateVariableOfTypeTime.lf
@@ -40,3 +40,4 @@ main reactor {
     =}
   }
 }
+

--- a/test/C/src/modal_models/ResetStateVariableOfTypeTime.lf
+++ b/test/C/src/modal_models/ResetStateVariableOfTypeTime.lf
@@ -40,4 +40,3 @@ main reactor {
     =}
   }
 }
-

--- a/test/C/src/modal_models/ResetStateVariableOfTypeTime.lf
+++ b/test/C/src/modal_models/ResetStateVariableOfTypeTime.lf
@@ -1,0 +1,45 @@
+/** 
+ * Modal Reactor Test. 
+ * Tests state variable with type time.
+ * https://github.com/lf-lang/lingua-franca/issues/1938
+ * Model by Edward Lee.
+ */
+target C {
+    timeout: 3s,
+    fast: true
+}
+
+reactor C {
+  input trigger:bool
+  reset state t:time = 0s
+  
+  reaction(trigger) {=
+    lf_print("t = %ld", self->t);
+    if (self->t != SEC(0)) {
+        lf_print("Error: Missing reset");
+    }
+    
+    self->t = lf_time_logical();
+  =}
+}
+
+main reactor {
+  timer t(0, 1s)
+  
+  initial mode A {
+    c = new C()
+    
+    reaction(t) -> reset(B), c.trigger {=
+      lf_print("In A");
+      lf_set(c.trigger, true);
+      lf_set_mode(B);
+    =}
+  }
+  
+  mode B {
+    reaction(t) -> reset(A) {=
+      lf_print("In B");
+      lf_set_mode(A);
+    =}
+  }
+}

--- a/test/C/src/modal_models/ResetStateVariableOfTypeTime.lf
+++ b/test/C/src/modal_models/ResetStateVariableOfTypeTime.lf
@@ -1,41 +1,38 @@
-/** 
- * Modal Reactor Test. 
- * Tests state variable with type time.
+/**
+ * Modal Reactor Test. Tests state variable with type time. Model by Edward Lee.
  * https://github.com/lf-lang/lingua-franca/issues/1938
- * Model by Edward Lee.
  */
 target C {
-    timeout: 3s,
-    fast: true
+  timeout: 3 s,
+  fast: true
 }
 
 reactor C {
-  input trigger:bool
-  reset state t:time = 0s
-  
+  input trigger: bool
+  reset state t: time = 0 s
+
   reaction(trigger) {=
     lf_print("t = %ld", self->t);
     if (self->t != SEC(0)) {
         lf_print("Error: Missing reset");
     }
-    
+
     self->t = lf_time_logical();
   =}
 }
 
 main reactor {
-  timer t(0, 1s)
-  
+  timer t(0, 1 s)
+
   initial mode A {
     c = new C()
-    
     reaction(t) -> reset(B), c.trigger {=
       lf_print("In A");
       lf_set(c.trigger, true);
       lf_set_mode(B);
     =}
   }
-  
+
   mode B {
     reaction(t) -> reset(A) {=
       lf_print("In B");

--- a/test/C/src/modal_models/ResetStateVariableWithParameterizedValue.lf
+++ b/test/C/src/modal_models/ResetStateVariableWithParameterizedValue.lf
@@ -1,41 +1,38 @@
-/** 
- * Modal Reactor Test. 
- * Tests state variable initialized via parameter.
+/**
+ * Modal Reactor Test. Tests state variable initialized via parameter. Model by Edward Lee.
  * https://github.com/lf-lang/lingua-franca/issues/1938
- * Model by Edward Lee.
  */
 target C {
-    timeout: 3s,
-    fast: true
+  timeout: 3 s,
+  fast: true
 }
 
-reactor C (init:int = 0) {
-  input trigger:bool
-  reset state i:int = init
-  
+reactor C(init: int = 0) {
+  input trigger: bool
+  reset state i: int = init
+
   reaction(trigger) {=
     lf_print("i = %d", self->i);
     if (self->i != -1) {
         lf_print("Error: Missing reset");
     }
-    
+
     self->i += 10;
   =}
 }
 
 main reactor {
-  timer t(0, 1s)
-  
+  timer t(0, 1 s)
+
   initial mode A {
-    c = new C(init = -1)
-    
+    c = new C(init=-1)
     reaction(t) -> reset(B), c.trigger {=
       lf_print("In A");
       lf_set(c.trigger, true);
       lf_set_mode(B);
     =}
   }
-  
+
   mode B {
     reaction(t) -> reset(A) {=
       lf_print("In B");

--- a/test/C/src/modal_models/ResetStateVariableWithParameterizedValue.lf
+++ b/test/C/src/modal_models/ResetStateVariableWithParameterizedValue.lf
@@ -40,3 +40,4 @@ main reactor {
     =}
   }
 }
+

--- a/test/C/src/modal_models/ResetStateVariableWithParameterizedValue.lf
+++ b/test/C/src/modal_models/ResetStateVariableWithParameterizedValue.lf
@@ -1,0 +1,45 @@
+/** 
+ * Modal Reactor Test. 
+ * Tests state variable initialized via parameter.
+ * https://github.com/lf-lang/lingua-franca/issues/1938
+ * Model by Edward Lee.
+ */
+target C {
+    timeout: 3s,
+    fast: true
+}
+
+reactor C (init:int = 0) {
+  input trigger:bool
+  reset state i:int = init
+  
+  reaction(trigger) {=
+    lf_print("i = %d", self->i);
+    if (self->i != -1) {
+        lf_print("Error: Missing reset");
+    }
+    
+    self->i += 10;
+  =}
+}
+
+main reactor {
+  timer t(0, 1s)
+  
+  initial mode A {
+    c = new C(init = -1)
+    
+    reaction(t) -> reset(B), c.trigger {=
+      lf_print("In A");
+      lf_set(c.trigger, true);
+      lf_set_mode(B);
+    =}
+  }
+  
+  mode B {
+    reaction(t) -> reset(A) {=
+      lf_print("In B");
+      lf_set_mode(A);
+    =}
+  }
+}

--- a/test/C/src/modal_models/ResetStateVariableWithParameterizedValue.lf
+++ b/test/C/src/modal_models/ResetStateVariableWithParameterizedValue.lf
@@ -40,4 +40,3 @@ main reactor {
     =}
   }
 }
-

